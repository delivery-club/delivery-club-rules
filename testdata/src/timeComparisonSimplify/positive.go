package timeComparisonSimplify

import (
	"time"
)

func _(x, y time.Time) {
	if !x.Before(y) { // want `\Qsuggestion: x.After(y)`
		print(42)
	}

	if !y.After(x) { // want `\Qsuggestion: y.Before(x)`
		print(51)
	}

	print(!y.Before(x)) // want `\Qsuggestion: y.After(x)`
}
