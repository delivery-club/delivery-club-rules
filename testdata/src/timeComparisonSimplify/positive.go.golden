package timeComparisonSimplify

import (
	"time"
)

func _(x, y time.Time) {
	if x.After(y) { // want `\Qsuggestion: x.After(y)`
		print(42)
	}

	if y.Before(x) { // want `\Qsuggestion: y.Before(x)`
		print(51)
	}

	print(y.After(x)) // want `\Qsuggestion: y.After(x)`
}
